dash/include/dash/memory/GlobHeapCombinedMem.h

```diff
@@ -0,0 +1,333 @@
+            }
+            _bucket_cumul_sizes[i][j + offset] += last;
+          }
+        }
+        ++offset;
+      }
+    }
+  }
+
+  /**
+   * Updates the bucket list.
+   */
+  void update_bucket_list() {
+    _buckets.clear();
+    std::vector<std::pair<typename bucket_list::iterator, 
+      typename bucket_list::iterator>> iters;
+    int max = 0;
+    if(_glob_mem_list.size() > 0) {
+      for(auto gmem : _glob_mem_list) {
+        iters.push_back(std::make_pair(gmem->_buckets.begin(), 
+              gmem->_buckets.end()));
+        if(max < gmem->_buckets.size()) {
+          max = gmem->_buckets.size();
+        }
+      }
+      // TODO: Currently onlys working for GlobMemContiguousMem because of
+      //       the double buckets
+      for(int i = 0; i < max; i += 2) {
+        for(auto & it : iters) {
+          if(it.first != it.second) {
+            _buckets.push_back(*(it.first));
+            ++(it.first);
+          }
+          if(it.first != it.second) {
+            _buckets.push_back(*(it.first));
+            ++(it.first);
+          }
+        }
+      }
+    }
+    update_local_size();
+    update_lbegin();
+    update_lend();
+  }
+
+private:
+
+  /**
+   * Updates the global size.
+   */
+  void update_size() {
+    _size = 0;
+    for(auto gmem : _glob_mem_list) {
+      _size += gmem->_size;
+    }
+  }
+
+  /**
+   * Updates the local size.
+   */
+  void update_local_size() {
+    _local_size = 0;
+    for(auto gmem : _glob_mem_list) {
+      _local_size += gmem->_local_size;
+    }
+  }
+
+  // NOTE: copied from GlobHeapMem.h
+  /**
+   * Native pointer of the initial address of the local memory of
+   * a unit.
+   *
+   */
+  void update_lbegin() noexcept
+  {
+    local_iterator unit_lbegin(
+             // iteration space
+             _buckets.begin(), _buckets.end(),
+             // position in iteration space
+             0,
+             // bucket at position in iteration space,
+             // offset in bucket
+             _buckets.begin(), 0);
+    _lbegin = unit_lbegin;
+  }
+
+  // NOTE: copied from GlobHeapMem.h
+  /**
+   * Update internal native pointer of the final address of the local memory
+   * of a unit.
+   */
+  void update_lend() noexcept
+  {
+    local_iterator unit_lend(
+             // iteration space
+             _buckets.begin(), _buckets.end(),
+             // position in iteration space
+             _local_size,
+             // bucket at position in iteration space,
+             // offset in bucket
+             _buckets.end(), 0);
+    _lend = unit_lend;
+  }
+
+private:
+
+  bucket_list              _buckets;
+  bucket_cumul_sizes_map   _bucket_cumul_sizes;
+  glob_mem_list            _glob_mem_list;
+  Team *                   _team = nullptr;
+  size_type                _size = 0;
+  global_iterator          _begin;
+  global_iterator          _end;
+  local_iterator           _lbegin;
+  local_iterator           _lend;
```

Comment:
The iterators are in an undefined state before the first `commit` call. Is this acceptable?

Reply:
Of course not, good catch.